### ISSUE TITLE
[dev.icinga.com #8438] DB IDO {host,service}checks command_line value is "Object of type 'icinga::Array'"

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga2#2618]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/8438

**Created by jnedila on 2015-02-13 20:14:25 +00:00**

Assignee: _mfriedrich_
Status: _Resolved_  (closed on _2015-02-14 22:50:03 +00:00_)
Target Version: _2.3.0_
Last Update: _2015-02-20 16:09:48 +00:00 (in Redmine)_

    Icinga Version: 2.2.4
    Backport?: Not yet backported
    Include in Changelog: 1

---

This is using Postgresql (we've tried 9.1 and 9.4)
The hostchecks and servicechecks table have a column called 'command\_line' which is being filled in with the value: **Object of type 'icinga::Array'**
Table definitions are below.

Sample entries:

    monitoring-production=# select * from icinga_servicechecks;
     servicecheck_id | instance_id | service_object_id | check_type | current_check_attempt | max_check_attempts | state | state_type |       start_time       | start_time_usec |        end_time        | end_time_usec | command_object_id | command_args |          command_line          | timeout | early_timeout |  execution_time  | latency | return_code |                       output                       | long_output |                           perfdata                            | endpoint_object_id 
    -----------------+-------------+-------------------+------------+-----------------------+--------------------+-------+------------+------------------------+-----------------+------------------------+---------------+-------------------+--------------+--------------------------------+---------+---------------+------------------+---------+-------------+----------------------------------------------------+-------------+---------------------------------------------------------------+--------------------
                   1 |           1 |                54 |          0 |                     1 |                  3 |     0 |          1 | 2015-02-12 20:11:16-05 |          773005 | 2015-02-12 20:11:20-05 |        884468 |                26 |              | *Object of type 'icinga::Array'* |       0 |             0 | 4.11146283149719 |       0 |           0 | PING OK - Packet loss = 0%, RTA = 44.78 ms         |             | rta=44.775002ms;100.000000;200.000000;0.000000 pl=0%;5;15;0   |                  1
                   2 |           1 |                54 |          0 |                     1 |                  3 |     0 |          1 | 2015-02-12 20:12:16-05 |          788628 | 2015-02-12 20:12:20-05 |        916216 |                26 |              | *Object of type 'icinga::Array'* |       0 |             0 | 4.12758803367615 |       0 |           0 | PING OK - Packet loss = 0%, RTA = 50.28 ms         |             | rta=50.276001ms;100.000000;200.000000;0.000000 pl=0%;5;15;0   |                  1
                   3 |           1 |                54 |          0 |                     1 |                  3 |     0 |          1 | 2015-02-12 20:13:16-05 |          722125 | 2015-02-12 20:13:20-05 |        783294 |                26 |              | *Object of type 'icinga::Array'* |       0 |             0 | 4.06116914749146 |       0 |           0 | PING OK - Packet loss = 0%, RTA = 42.91 ms         |             | rta=42.912998ms;100.000000;200.000000;0.000000 pl=0%;5;15;0   |                  1
                   4 |           1 |                54 |          0 |                     1 |                  3 |     0 |          1 | 2015-02-12 20:14:16-05 |          781088 | 2015-02-12 20:14:20-05 |        901276 |                26 |              | *Object of type 'icinga::Array'* |       0 |             0 | 4.12018799781799 |       0 |           0 | PING OK - Packet loss = 0%, RTA = 41.29 ms         |             | rta=41.292999ms;100.000000;200.000000;0.000000 pl=0%;5;15;0   |                  1

                                                  Table "public.icinga_hostchecks"
            Column         |           Type           |                                Modifiers                                 
    -----------------------+--------------------------+--------------------------------------------------------------------------
     hostcheck_id          | bigint                   | not null default nextval('icinga_hostchecks_hostcheck_id_seq'::regclass)
     instance_id           | bigint                   | default 0
     host_object_id        | bigint                   | default 0
     check_type            | integer                  | default 0
     is_raw_check          | integer                  | default 0
     current_check_attempt | integer                  | default 0
     max_check_attempts    | integer                  | default 0
     state                 | integer                  | default 0
     state_type            | integer                  | default 0
     start_time            | timestamp with time zone | default '1970-01-01 00:00:00-05'::timestamp with time zone
     start_time_usec       | integer                  | default 0
     end_time              | timestamp with time zone | default '1970-01-01 00:00:00-05'::timestamp with time zone
     end_time_usec         | integer                  | default 0
     command_object_id     | bigint                   | default 0
     command_args          | text                     | default ''::text
     command_line          | text                     | default ''::text
     timeout               | integer                  | default 0
     early_timeout         | integer                  | default 0
     execution_time        | double precision         | default 0
     latency               | double precision         | default 0
     return_code           | integer                  | default 0
     output                | text                     | default ''::text
     long_output           | text                     | default ''::text
     perfdata              | text                     | default ''::text
     endpoint_object_id    | bigint                   | 

                                                    Table "public.icinga_servicechecks"
            Column         |           Type           |                                   Modifiers                                    
    -----------------------+--------------------------+--------------------------------------------------------------------------------
     servicecheck_id       | bigint                   | not null default nextval('icinga_servicechecks_servicecheck_id_seq'::regclass)
     instance_id           | bigint                   | default 0
     service_object_id     | bigint                   | default 0
     check_type            | integer                  | default 0
     current_check_attempt | integer                  | default 0
     max_check_attempts    | integer                  | default 0
     state                 | integer                  | default 0
     state_type            | integer                  | default 0
     start_time            | timestamp with time zone | default '1970-01-01 00:00:00-05'::timestamp with time zone
     start_time_usec       | integer                  | default 0
     end_time              | timestamp with time zone | default '1970-01-01 00:00:00-05'::timestamp with time zone
     end_time_usec         | integer                  | default 0
     command_object_id     | bigint                   | default 0
     command_args          | text                     | default ''::text
     command_line          | text                     | default ''::text
     timeout               | integer                  | default 0
     early_timeout         | integer                  | default 0
     execution_time        | double precision         | default 0
     latency               | double precision         | default 0
     return_code           | integer                  | default 0
     output                | text                     | default ''::text
     long_output           | text                     | default ''::text
     perfdata              | text                     | default ''::text
     endpoint_object_id    | bigint                   | 

**Changesets**

_2015-02-14 22:46:55 +00:00_ by _mfriedrich_ c8fe61010b7e4e8069cbb545b47b4106e39996b7

    DB IDO: Fix {host,service}checks command_line value is "Object of type 'icinga::Array'"
    
    fixes #8438

_2015-02-14 23:24:05 +00:00_ by _mfriedrich_ 4a64d4991b108b09da401afca8cce02e3c6aca16

    DB IDO: Fix {host,service}checks command_line value is "Object of type 'icinga::Array'"
    
    fixes #8438
